crates/parser/src/grammar/items.rs

```diff
@@ -389,10 +393,12 @@ fn macro_rules(p: &mut Parser, m: Marker) {
     }
 
     match p.current() {
-        T!['{'] => {
+        T!['['] | T!['('] => {
             token_tree(p);
+            p.expect(T![;]);
```

Comment:
Huh, TIL.

Reply:
Only `{}`, `[];` and `();` macro invocations are allowed to produce items, hence you also need the semi here.